scripts/azdo/update_sstate_cache.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+set -e
+
+usage() {
+	local exit_code=${1:-2}
+	test $exit_code -eq 0 || exec 1>&2
+
+	cat <<EOF
+
+Syntax:
+$(basename $BASH_SOURCE) push local_sstate_cache_dir argo_sstate_cache_dir
+   Push the local sstate-cache directory contents to the remote argo
+   server directory. Note that this script assumes sequential pushes
+   to the cache.
+
+$(basename $BASH_SOURCE) pull local_sstate_cache_dir argo_sstate_cache_dir
+   Fetch the latest sstate cache from the remote argo cache directory
+   and install to the local sstate-cache directory.
+
+$(basename $BASH_SOURCE) -h
+   Print this help message and exit
+
+Arguments:
+local_sstate_cache_dir      Path to the local sstate cache directory
+argo_sstate_cache_dir       Path to the remote argo sstate cache directory
+
+EOF
+	exit $exit_code
+}
+
+push_cache() {
+	if [ ! -d "$local_sstate_cache_dir" -o -z "$(ls -A $local_sstate_cache_dir 2>/dev/null)" ]; then
+		echo $local_sstate_cache_dir is missing/empty. Please provide a valid local_sstate_cache_dir name with sstate-cache contents.
+		usage
+	fi
+
+	mkdir -p "$argo_sstate_cache_dir"
+	pushd "$argo_sstate_cache_dir" >/dev/null
+
+	# Copy the contents of the cache to a temp .latest directory
+	# before removing the .latest suffix from the name. Create a
+	# DONOTUSE file in the .latest directory until the mv operation is
+	# complete. The pull operation will not rsync a folder with this
+	# file. Implement this extra precaution because the mv may not be
+	# atomic on the argo share.
+
+	temp_argo_dir=$(date +"%s").latest
+	trap "{ rm -rf $temp_argo_dir; }" EXIT
```

Comment:
`TRAP`s have scope until the process exits, so this `TRAP` will be in place for the entirety of this script. It's probably inconsequential - since `$temp_argo_dir` should not exist for very long anyway - but it would be cleaner if you unset the TRAP after the target directory gets renamed (line 49). You can use the provisioning test script I wrote recently [as a reference](https://github.com/ni/nilrt/blob/nilrt/master/dunfell/scripts/tests/provisioning/start-vm-using-expect.sh#L126) for how to do that.